src/ui/common/src/components/workflows/artifact/check/history.tsx

```diff
@@ -51,14 +51,21 @@ const CheckHistory: React.FC<CheckHistoryProps> = ({
     schema: checkHistorySchema,
     data: (historyWithLoadingStatus.results?.results ?? []).map(
       (artifactStatusResult) => {
-        let timestamp = new Date(
-          artifactStatusResult.exec_state?.timestamps?.finished_at
-        ).toLocaleString();
+        const all_times = [
+          artifactStatusResult.exec_state?.timestamps?.finished_at,
+          artifactStatusResult.exec_state?.timestamps?.pending_at,
+          artifactStatusResult.exec_state?.timestamps?.registered_at,
+          artifactStatusResult.exec_state?.timestamps?.running_at,
+        ];
 
-        // Checks that are canceled / fail to execute have no exec_state or finished_at time.
-        if (timestamp === 'Invalid Date') {
-          timestamp = 'Unknown';
-        }
+        const timesOrNull = all_times.map((x) =>
+          typeof x === 'string' ? new Date(x) : null
+        );
+
+        const maxTime = Math.max.apply(null, timesOrNull);
+
+        const timestamp =
+          maxTime > 0 ? new Date(maxTime).toLocaleString() : 'Unknown';
```

Comment:
If we following the suggested change in line 61, we can assign this value based on if the list is empty